app/src/main/java/org/mozilla/focus/home/HomeFragment.kt

```diff
@@ -90,6 +92,22 @@ class HomeFragment : Fragment() {
         }
 
         registerForContextMenu(view)
+
+        tileContainer.addOnScrollListener(object : RecyclerView.OnScrollListener() {
+            override fun onScrolled(recyclerView: RecyclerView?, dx: Int, dy: Int) {
+                if (dy > 0) {
+                    val layoutManager = recyclerView?.layoutManager as GridLayoutManager
+                    val visibleItemCount = layoutManager.childCount
```

Comment:
Is this actually correct? Can you explain to me why this would work for what we want it to do, and does it work for all cases?

I'm not stoked about code from Stackoverflow, without comments or explanation...please add code comments for why this works and what problem this is fixing, or at least a comment in the github.

Ideally the goal for code (or hacks) is to make it easy to understand, optionally with comments detailing when/why this works (if the logic isn't simple/obvious). That way, if it introduces bugs, anyone could glance at the code and see "oh this is a wrong assumption" and have a fix.